examples/fast_neural_style/neural_style.py

```diff
@@ -0,0 +1,291 @@
+    mse_loss = torch.nn.MSELoss()
+
+    vgg = Vgg16(requires_grad=False).to(device)
+    style_transform = transforms.Compose([
+        transforms.ToTensor(),
+        transforms.Lambda(lambda x: x.mul(255))
+    ])
+    if args.style_image != 'test':
+        style = utils.load_image(args.style_image, size=args.style_size)
+    else:
+        imarray = np.random.rand(32, 32, 3).astype('uint8')
+        style = Image.fromarray(imarray)
+
+    style = style_transform(style)
+
+    style = style.repeat(args.batch_size, 1, 1, 1).to(device)
+
+    features_style = vgg(utils.normalize_batch(style))
+    gram_style = [utils.gram_matrix(y) for y in features_style]
+
+    running_avgs = OrderedDict()
+    output_stream = sys.stdout
+    alpha = 0.98
+
+    def step(engine, batch):
+
+        x, _ = batch
+        x = x.to(device)
+
+        n_batch = len(x)
+
+        transformer.zero_grad()
+
+        y = transformer(x)
+
+        x = utils.normalize_batch(x)
+        y = utils.normalize_batch(y)
+
+        features_x = vgg(x)
+        features_y = vgg(y)
+
+        content_loss = args.content_weight * mse_loss(features_y.relu2_2, features_x.relu2_2)
+
+        style_loss = 0.
+        for ft_y, gm_s in zip(features_y, gram_style):
+            gm_y = utils.gram_matrix(ft_y)
+            style_loss += mse_loss(gm_y, gm_s[:n_batch, :, :])
+        style_loss *= args.style_weight
+
+        total_loss = content_loss + style_loss
+        total_loss.backward()
+        optimizer.step()
+
+        return {
+            'content_loss': content_loss.item(),
+            'style_loss': style_loss.item(),
+            'total_loss': total_loss.item()
+        }
+
+    trainer = Engine(step)
+    checkpoint_handler = ModelCheckpoint(args.checkpoint_model_dir, 'checkpoint',
+                                         save_interval=args.checkpoint_interval,
+                                         n_saved=10, require_empty=False, create_dir=True)
+
+    @trainer.on(Events.ITERATION_COMPLETED)
+    def update_logs(engine):
+        for k, v in engine.state.output.items():
+            old_v = running_avgs.get(k, v)
+            new_v = alpha * old_v + (1 - alpha) * v
+            running_avgs[k] = new_v
+
+    @trainer.on(Events.ITERATION_COMPLETED)
+    def print_logs(engine):
+
+        num_seen = engine.state.iteration - len(train_loader) * (engine.state.epoch - 1)
+
+        percent_seen = 100 * (float(num_seen / len(train_loader)))
+        percentages = list(range(0, 110, 10))
+
+        if int(percent_seen) == 100:
+            progress = 0
+            equal_to = 10
+            sub = 0
+
+        else:
+            sub = 1
+            progress = 1
+            equal_to = np.max(np.where([percent < percent_seen for percent in percentages])[0])
+
+        bar = '[' + '=' * equal_to + '>' * progress + ' ' * (10 - equal_to - sub) + ']'
+
+        message = 'Epoch {epoch} | {percent_seen:.2f}% | {bar}'.format(epoch=engine.state.epoch,
```

Comment:
Concerning your progress bar, looks cool on the execution. Could you please refactor it in a separate function ? Such that it could be reused by users if they would like to

Reply:
Added handlers.py with Progbar class. The code is now a lot cleaner. And I was wondering if it could be added to the ignite handlers? Maybe that's a separate issue.

Reply:
@anmolsjoshi Look good ! Yes, I thought about to propose a sort of RunningMetrics handler. However in this case we would need to separate running metric computing and progress bar logic. For a progress bar reporting, maybe we could take a look at tqdm direction.